topdown/crypto.go

```diff
@@ -40,8 +44,56 @@ func builtinCryptoX509ParseCertificates(a ast.Value) (ast.Value, error) {
 	return ast.InterfaceToValue(x)
 }
 
+func builtinCryptoMd5(a ast.Value) (ast.Value, error) {
+
+	s, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	md5 := md5.Sum([]byte(s))
+
+	stringMd5 := fmt.Sprintf("%x", md5)
+
+	return ast.String(stringMd5), nil
+
+}
+
+func builtinCryptoSha1(a ast.Value) (ast.Value, error) {
+
+	s, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	sha1 := sha1.Sum([]byte(s))
+
+	stringSha1 := fmt.Sprintf("%x", sha1)
+
+	return ast.String(stringSha1), nil
+
+}
+
+func builtinCryptoSha256(a ast.Value) (ast.Value, error) {
+
+	s, err := builtins.StringOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	sha256 := sha256.Sum256([]byte(s))
+
+	stringSha256 := fmt.Sprintf("%x", sha256)
+
+	return ast.String(stringSha256), nil
```

Comment:
[nit] I'm not sure if its worth it, but couldn't these be deduplicated a bit? Like,

```go
func hashHelper(a ast.Value, h func(string) string) (ast.Value, error) {
	s, err := builtins.StringOperand(a, 1)
	if err != nil {
		return nil, err
	}
	return ast.String(h(s)), nil
}

func builtinCryptoSha256(a ast.Value) (ast.Value, error) {
	return hashHelper(a, func(s string) { return fmt.Sprintf("%x", sha256.Sum256([]byte(s))) })
}

// etc
```

Reply:
Done and commits squashed.